### ISSUE TITLE
Document our approach to accessibility

### ISSUE THREAD

**Issue by laurent22** [laurent22/joplin#11860]
We should create a document to describe our approach in terms of accessibility. The document will drive development in general and ensures we don't introduce regressions over time and keep the apps (desktop and mobile) accessible.

The reader for this document would be a new developer who wants to contribute to Joplin. Because without knowing about accessibility it's almost certain that new developments will introduce regressions. May be particularly relevant if we participate in GSoC again, or even for ourselves as a way to review we took everything into consideration when adding a new feature.

The document should probably contain information such as:

- What should be taken into consideration when creating new components
    - Mention ARIA labels
    - Keyboard shortcut
    - Focus issues
    - Announcements
    - And more - review https://github.com/laurent22/joplin/issues/10795 and add any relevant item
- How can we verify that a change or new feature is accessible
    - Provide links to simple guides on how to setup screen readers on various platforms
    - Any other way to check?
- How to add tests for accessibility to prevent regressions
    - Are there tests or eslint rules we can add to ensure that new components for example are accessible?
    - And describe how to manually add tests, if that's possible (maybe just linking to our existing test doc if that's sufficient)
- Anything else?

WCAG 2 is massive obviously so we can't document everything so we should focus on the most important parts, so as to create a document that can be read in a relatively short time. As additional references, we can link to WCAG 2, WebAIM contrast checker, and other useful tools and documents.

**Comment by personalizedrefrigerator** [comment]
I've started working on this! Regarding ESLint rules, perhaps we could add [eslint-plugin-react-native-a11y](https://github.com/FormidableLabs/eslint-plugin-react-native-a11y).
